### PR TITLE
Fix audit, update to go version 19

### PR DESCRIPTION
**Pull request by redhug1**
### What

fix audit, update version of go to 19

### How to review

run audit test
run other tests

### Who can review

Anyone

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,9 +1,13 @@
 module github.com/ONSdigital/dp-cantabular-dimension-api
 
-go 1.17
+go 1.19
 
 replace github.com/coreos/etcd => github.com/coreos/etcd v3.3.24+incompatible
 
+replace github.com/prometheus/client_golang => github.com/prometheus/client_golang v1.11.1
+
+replace golang.org/x/text => golang.org/x/text v0.3.8
```
Comment (rafahop) [inline go.mod]: I think we wouldn't need this directive if we ran `go get golang.org/x/text@latest`?

File: go.mod [hunk 2]
```diff
@@ -1,9 +1,13 @@
 module github.com/ONSdigital/dp-cantabular-dimension-api
 
-go 1.17
+go 1.19
 
 replace github.com/coreos/etcd => github.com/coreos/etcd v3.3.24+incompatible
 
+replace github.com/prometheus/client_golang => github.com/prometheus/client_golang v1.11.1
```
Comment (rafahop) [inline go.mod]: Should we use the latest version?

**Review by rafahop — COMMENTED** [review]
Not something you've introduced but we could take the opportunity to join the 2 groups of indirect dependencies in `go.mod` and keep them in one. 
Not sure why it's been split in 2 but probably because `github.com/gorilla/schema` is actually a direct dependency living in the indirect require group. If you could manually move that up to the "direct" group and join the 2 other require groups, then running `go.mod` would tidy it up 🙂

File: go.mod (hunk 1 above)
Comment (redhug1) [reply go.mod]: I tried what you suggested and that broke the audit in CI

So, i've had to put it back.

File: go.mod (hunk 2 above)
Comment (redhug1) [reply go.mod]: done

File: go.mod (hunk 1 above)
Comment (redhug1) [reply go.mod]: Now fixed

**Review by sandypadmanabhan — APPROVED** [review]

**Pull request merged**